include/bm/bm_sim/named_p4object.h

```diff
@@ -46,9 +51,13 @@ class NamedP4Object {
   //! Get the compiler-provided id
   p4object_id_t get_id() const { return id; }
 
+  bool has_source_info() const { return (source_info != nullptr); }
+  const SourceInfo *get_source_info() const { return source_info; }
+
  protected:
   const std::string name;
   p4object_id_t id;
+  const SourceInfo *source_info;
```

Comment:
I have attempted to change the SourceInfo member of class NamedP4Object to use unique_ptr, but I freely admit that C++'s rules for these things have stymied me.

I have checked in some code that attempts to use unique_ptr for this, on a branch that is not part of this PR, here: https://github.com/jafingerhut/behavioral-model/commit/3c7f2411d480cf984ce0fd7b4b4c48c51f222cac

I have added a comment to my own diff there with some questions, in case anyone is interested in taking a look -- if you know C++ default copy constructors and how they interact with unique_ptr fields, you may know what I am missing here.